modules/codegen/src/smithy4s/codegen/Renderer.scala

```diff
@@ -39,27 +40,63 @@ object Renderer {
     val pack = Result(
       unit.namespace,
       "package",
-      r.renderPackageContents.lines.mkString(
-        System.lineSeparator()
-      )
+      r.renderPackageContents.list
+        .map(_.segments.toList.map(_.show).mkString)
+        .mkString(
+          System.lineSeparator()
+        )
     )
 
     val classes = unit.declarations.map { decl =>
       val renderResult = r.renderDecl(decl)
       val p = s"package ${unit.namespace}"
 
-      val allImports =
-        renderResult.imports.filter(
-          _.replaceAll(unit.namespace, "")
-            .split('.')
-            .count(_.nonEmpty) > 1
-        )
+      val nameCollisions: Set[String] = renderResult.list
+        .flatMap(_.segments.toList)
+        .distinct
+        .collect {
+          case NameRef(_, name) => name
+          case NameDef(name)    => name
+        }
+        .groupBy(identity)
+        .filter(_._2.size > 1)
+        .keySet
+
+      /*      def condenseSmithy4sImports(
+          typeReference: TypeReference
+      ): TypeReference = {
+        typeReference match {
+          case tr @ TypeReference(List("smithy4s"), _) => tr.copy(name = "_")
+          case tr @ TypeReference(List("smithy4s", "schema", "Schema"), _) =>
+            tr.copy(name = "_")
+          case tr => tr
+        }
+      }*/
```

Comment:
Is this comment left in on purpose?

Reply:
no , thought I pushed that change, pushing now